pkg/systemd/terminal.jsx

```diff
@@ -103,7 +101,9 @@ const _ = cockpit.gettext;
                 }
             }
             const user = await cockpit.user();
-            this.setState({ user, channel: this.createChannel(user, dir) });
+            this.setState({ user, channel: this.createChannel(user, dir) },
+                          () => { cockpit.addEventListener("locationchanged", this.onNavigate) }
+            );
```

Comment:
If there is a navigation event before user gets initialized, it would now just be ignored (instead of crashing). That's better than crashing, but `onNavigate()` is already async -- could this not do `await cockpit.user()`? Does that crash because `prevState.user` is undefined? Could that be replaced with the direct `await cockpit.user()` value?